cJSON.c

```diff
@@ -2057,6 +2057,124 @@ CJSON_PUBLIC(cJSON_bool) cJSON_AddItemToObjectCS(cJSON *object, const char *stri
     return add_item_to_object(object, string, item, &global_hooks, true);
 }
 
+CJSON_PUBLIC(void) cJSON_AddItemReferenceToObjectCS(cJSON *object, const char *string, cJSON *item)
```

Comment:
We generally try to avoid `toup` patches unless it's an emergency and there really is no time to submit upstream first.

Can you please:

1. submit this patch upstream (without any `[nrf ...]` tag, of course)
2. change this commit to `fromlist`
3. add a link to the upstream PR in the commit log of the downstream `fromlist` patch?

If upstream rejects it, we can carry a `noup`. If upstream is slow to respond, we can carry a `fromlist`. But let's try to get this upstreamed, please.

Reply:
I missed this partner PR to https://github.com/nrfconnect/sdk-nrf/pull/4409 entirely.  Sorry about that.

Yes, I'll submit my changes upstream when I am less busy with the big push to get predicted GPS into NCS 1.6.0.  Hope more delay is acceptable.